***Άσκηση plus 1.*** Ζητείται να γραφτεί ένα πρόγραμμα προσομοίωσης μιας απλής αριθμομηχανής. Ζητείται από τον χρήστη να ορίσει δυο αριθμούς και μια αριθμητική πράξη. Επιτρεπόμενες αριθμητικές πράξεις είναι οι εξής: πρόσθεση, αφαίρεση, πολλαπλασιασμός, διαίρεση, εύρεση ακέραιου και υπόλοιπου πηλίκου διαίρεσης. 

In [3]:
def arithmetic_operation(num1, num2, operator):
    if operator == "+":
        return num1 + num2
    elif operator == "-":
        return num1 - num2
    elif operator == "*":
        return num1 * num2
    elif operator == "/":
        if num2 != 0:
            return num1 / num2
        else:
            return "Error: division by zero"
    elif operator == "%":
        if num2 != 0:
            return num1 % num2
        else:
            return "Error: modulo by zero"
    else:
        return "Error: invalid operator"


# Take input from the user
num1 = float(input("Enter first number: "))
operator = input("What you want to do (+, -, *, /, %): ")
num2 = float(input("Enter second number: "))

result = arithmetic_operation(num1, num2, operator)

print(f"\n{num1} {operator} {num2} = {result}")


Enter first number: 3.456
What you want to do (+, -, *, /, %): *
Enter second number: 54.234

3.456 * 54.234 = 187.432704


Ο δεδομένος κώδικας ορίζει μια συνάρτηση που ονομάζεται arithmetic_operation() η οποία εκτελεί απλές αριθμητικές πράξεις σε δύο αριθμητικές τιμές num1 και num2 με βάση τον τελεστή που του παρέχεται ως τρίτος τελεστής ορίσματος.

Οι όροι if, elif και else χρησιμοποιούνται για να ελεγχθεί εάν ο τελεστής είναι έγκυρος ή όχι. Εάν δεν είναι έγκυρος τελεστής, η συνάρτηση επιστρέφει μια συμβολοσειρά σφάλματος "Error: invalid operator". Διαφορετικά, εκτελεί την αντίστοιχη αριθμητική πράξη με βάση τον τελεστή εισόδου.
Για τους τελεστές διαίρεσης (/) και modulo (%), η συνθήκη if num2 != 0: ελέγχεται πρώτα για να αποφευχθεί το ZeroDivisionError. Εάν το num2 είναι μηδέν για αυτούς τους τελεστές, τότε η συνάρτηση θα πρέπει να επιστρέψει ένα μήνυμα σφάλματος "Error: division by zero" ή "Error: modulo by zero" αντίστοιχα.
Μετά τον καθορισμό αυτής της συνάρτησης, ο κωδικός λαμβάνει την είσοδο του χρήστη για δύο αριθμούς num1 και num2 και έναν τελεστή χειριστή. Στη συνέχεια καλεί την καθορισμένη συνάρτηση arithmetic_operation() για να εκτελέσει τον υπολογισμό και να αποθηκεύσει το αποτέλεσμα σε ένα μεταβλητό αποτέλεσμα. Τέλος, εκτυπώνεται μια μορφοποιημένη συμβολοσειρά για να εμφανιστεί ολόκληρη η αριθμητική παράσταση μαζί με το αποτέλεσμά της.

***Άσκηση plus 2.*** Ζητείται να γραφτεί κώδικας που θα βρίσκει το άθροισμα μιας σειράς αριθμών που υπάρχουν σε μια λίστα. 

In [4]:
def get_sum(nums):
   sum = 0
   for num in nums:
       sum = sum + num
   return sum 
nums = [13,89,65,42,12,11,56] 
total = get_sum(nums)
print("The total of each elements:",total)

The total of each elements: 288


Ο κώδικας ορίζει μια συνάρτηση που ονομάζεται get_sum η οποία παίρνει μια λίστα ακεραίων αριθμών και επιστρέφει το άθροισμά τους.
Η μεταβλητή sum αρχικοποιείται στο 0 στην αρχή της συνάρτησης. Ένας βρόχος for επαναλαμβάνεται μέσω κάθε στοιχείου των αριθμών λίστας εισόδου.
Στη συνέχεια, ο τρέχων ακέραιος προστίθεται στην μεταβλητή sum και αυτή η διαδικασία επαναλαμβάνεται μέχρι να αθροιστούν όλα τα στοιχεία της λίστας. Τέλος, η συνάρτηση επιστρέφει το υπολογισμένο άθροισμα.
Μετά τον ορισμό της συνάρτησης, ο κώδικας δημιουργεί μια λίστα ακεραίων αριθμών που ονομάζεται nums και της εκχωρεί κάποιες τιμές. Καλεί τη συνάρτηση get_sum, περνώντας στη λίστα nums ως όρισμα. Η επιστρεφόμενη τιμή αποθηκεύεται στην μεταβλητή total.
Τέλος, ο κώδικας εκτυπώνει τη συμβολοσειρά "Το σύνολο των στοιχείων:" ακολουθούμενη από την τιμή του συνόλου. Σε αυτήν την περίπτωση, η έξοδος θα είναι "Το σύνολο κάθε στοιχείου: 288", καθώς το άθροισμα των τιμών σε αριθμούς είναι 288.

***Άσκηση plus 3.*** Βρείτε τον μέγιστο αριθμό δύο αριθμών.